From 4f05df59b794b941ad7f6295c4b4f906f33e25f4 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Mon, 26 Oct 2009 18:26:19 +0800
Subject: [PATCH] net/dpa: invoke FM_MAC_Restart function when starting dpa mac

We set the graceful stop bit when down the mac, so invoke
FM_MAC_Restart function to make sure we clear these graceful
stop bit when up the mac again. Otherwise the mac will stop
transmission even after we have enabled it.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/dpa/mac-api.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/net/dpa/mac-api.c b/drivers/net/dpa/mac-api.c
index c1bdf28..49014bb 100644
--- a/drivers/net/dpa/mac-api.c
+++ b/drivers/net/dpa/mac-api.c
@@ -193,6 +193,12 @@ static int __cold start(struct mac_device *mac_dev)
 		cpu_dev_err(mac_dev->dev, "%s:%hu:%s(): FM_MAC_Enable() = 0x%08x\n",
 			    __file__, __LINE__, __func__, err);
 
+	err = FM_MAC_Restart(((struct mac_priv_s *)macdev_priv(mac_dev))->mac,
+			e_COMM_MODE_RX_AND_TX);
+	_errno = -GET_ERROR_TYPE(err);
+	if (unlikely(_errno < 0))
+		cpu_dev_err(mac_dev->dev, "%s:%hu:%s(): FM_MAC_Restart() = 0x%08x\n",
+			    __file__, __LINE__, __func__, err);
 	return _errno;
 }
 
-- 
1.6.5.rc3

